HPX fails to build on macOS

HPX ee38b8bd657607a7e52587fcf3df6d06fb4a3480 fails to build on macOS 10.14.6, AppleClang 10.0.1

Error:
```console
+ /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x c++ -arch x86_64 -fmessage-length=195 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit=0 -fcolor-diagnostics -Wno-trigraphs -fpascal-strings -O0 -Wno-missing-field-initializers -Wno-missing-prototypes -Wno-return-type -Wno-non-virtual-dtor -Wno-overloaded-virtual -Wno-exit-time-destructors -Wno-missing-braces -Wparentheses -Wswitch -Wno-unused-function -Wno-unused-label -Wno-unused-parameter -Wno-unused-variable -Wunused-value -Wno-empty-body -Wno-uninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wno-constant-conversion -Wno-int-conversion -Wno-bool-conversion -Wno-enum-conversion -Wno-float-conversion -Wno-non-literal-null-conversion -Wno-objc-literal-conversion -Wno-shorten-64-to-32 -Wno-newline-eof -Wno-c++11-extensions -DCMAKE_INTDIR=\"Debug\" -DDEBUG -D_DEBUG -DHPX_MODULE_EXPORTS -D_GNU_SOURCE -DBOOST_ALL_NO_LIB -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk -fasm-blocks -fstrict-aliasing -Wdeprecated-declarations -Winvalid-offsetof -mmacosx-version-min=10.14 -g -Wno-sign-conversion -Wno-infinite-recursion -Wno-move -Wno-comma -Wno-block-capture-autoreleasing -Wno-strict-prototypes -Wno-range-loop-analysis -Wno-semicolon-before-method-body -I${REPOS}/hpx/cmake-build-debug/lib/Debug/include -I/usr/local/Cellar/hwloc/2.0.4/include -I${REPOS}/hpx/cmake-build-debug -I${REPOS}/hpx/libs/logging/include -I${REPOS}/hpx/cmake-build-debug/libs/logging/include -I${REPOS}/hpx/libs/logging/include_compatibility -I${REPOS}/hpx/libs/assertion/include -I${REPOS}/hpx/cmake-build-debug/libs/assertion/include -I${REPOS}/hpx/libs/assertion/include_compatibility -I${REPOS}/hpx/libs/config/include -I${REPOS}/hpx/cmake-build-debug/libs/config/include -I${REPOS}/hpx/libs/preprocessor/include -I${REPOS}/hpx/cmake-build-debug/libs/preprocessor/include -I${REPOS}/hpx/libs/preprocessor/include_compatibility -I${REPOS}/hpx/libs/filesystem/include -I${REPOS}/hpx/cmake-build-debug/libs/filesystem/include -isystem /usr/local/include -I${REPOS}/hpx/cmake-build-debug/libs/logging/HPX.build/Debug/hpx_logging.build/DerivedSources-normal/x86_64 -I${REPOS}/hpx/cmake-build-debug/libs/logging/HPX.build/Debug/hpx_logging.build/DerivedSources/x86_64 -I${REPOS}/hpx/cmake-build-debug/libs/logging/HPX.build/Debug/hpx_logging.build/DerivedSources -Wmost -Wno-four-char-constants -Wno-unknown-pragmas -F${REPOS}/hpx/cmake-build-debug/lib/Debug -fPIC -std=c++17 -pthread -ftemplate-depth=256 -Wall -Wextra -Wno-unused-local-typedefs -Wno-strict-aliasing -Wno-sign-promo -Wno-attributes -Wno-cast-align -Wno-unused-parameter -Wformat=2 -Wno-format-nonliteral -Winit-self -Wdouble-promotion -Wcast-qual -Wcast-align -Werror=parentheses -Werror=reorder -Werror=return-type -Werror=sequence-point -Werror=uninitialized -Werror=format -Werror=missing-braces -Werror=sign-compare -fdiagnostics-show-option -Werror=vla -MMD -MT dependencies -MF ${REPOS}/hpx/cmake-build-debug/libs/logging/HPX.build/Debug/hpx_logging.build/Objects-normal/x86_64/logging.d --serialize-diagnostics ${REPOS}/hpx/cmake-build-debug/libs/logging/HPX.build/Debug/hpx_logging.build/Objects-normal/x86_64/logging.dia -c ${REPOS}/hpx/libs/logging/src/logging.cpp -o ${REPOS}/hpx/cmake-build-debug/libs/logging/HPX.build/Debug/hpx_logging.build/Objects-normal/x86_64/logging.o
In file included from ${REPOS}/hpx/libs/logging/src/logging.cpp:11:
In file included from ${REPOS}/hpx/libs/logging/include/hpx/logging.hpp:17:
In file included from ${REPOS}/hpx/libs/logging/include/hpx/logging/logging.hpp:21:
In file included from ${REPOS}/hpx/libs/logging/include/hpx/logging/detail/logger.hpp:21:
In file included from ${REPOS}/hpx/libs/logging/include/hpx/logging/detail/cache_before_init.hpp:22:
In file included from ${REPOS}/hpx/libs/logging/include/hpx/logging/writer/named_write.hpp:29:
${REPOS}/hpx/libs/logging/include/hpx/logging/format/formatter/high_precision_time.hpp:34:10: fatal error: 'hpx/lcos/local/spinlock.hpp' file not found
#include <hpx/lcos/local/spinlock.hpp>
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Full log:
[build-fail.txt](https://github.com/STEllAR-GROUP/hpx/files/3570641/build-fail.txt)

--- a/cmake/HPX_AddModule.cmake
+++ b/cmake/HPX_AddModule.cmake
@@ -173,6 +173,12 @@ function(add_hpx_module name)
     HPX_MODULE_EXPORTS
   )
 
+  # This is a temporary solution until all of HPX has been modularized as it
+  # enables using header files from HPX for compiling this module.
+  target_include_directories(hpx_${name} PRIVATE
+    ${CMAKE_CURRENT_SOURCE_DIR}/../..
+  )
+
   add_hpx_source_group(
     NAME hpx_{name}
     ROOT ${HEADER_ROOT}/hpx

